### PR TITLE
[RFC] Script DNS provider

### PR DESCRIPTION
**Pull request by iavael**
Prototype of integration with any DNS systems via external script/binary

### CODE REVIEW

**Comment by mholt** [comment]
This is an interesting idea; so it looks like the script must conform to a certain CLI (with subcommands like present and cleanup). Is that a reasonable expectation in general?

**Comment by iavael** [comment]
@mholt

>  Is that a reasonable expectation in general?

In general - yes. These two "providers" are for custom integration (e.g. with internal dns management system), so it should be expected to write a short script.


I hasn't documented script and pipe interfaces yet, because I'm still working on it. I gonna put README files in respected directories.

File: providers/dns/exec/exec.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Package exec implements a DNS provider for solving the DNS-01 challenge using
+// a user-provided executable.
+package exec
```
Comment (mholt) [inline providers/dns/exec/exec.go]: Can the package be named something less "stdliby"? Like `dnsexec` maybe? Then we can ditch the renaming of the os/exec package.

**Comment by mholt** [comment]
Closing in favor of #508

**Pull request closed without merging**